src/libm/CMakeLists.txt

```diff
@@ -189,23 +197,90 @@ endif()
 # Note: adds object file as sources via cmake conditional generator expression
 foreach(SIMD ${SLEEF_SUPPORTED_EXTENSIONS})
   if(COMPILER_SUPPORTS_${SIMD})
+    list(FIND EXT_ENABLE_ALIAS ${SIMD} INDEX_ALIAS)
+    string(TOLOWER ${SIMD} SIMDLC)
+
+    # SP functions
+
     # Create a library
-    add_library(${OBJECT_${SIMD}} OBJECT ${SIMD_SOURCES} ${HEADER_${SIMD}})    
-    target_compile_definitions(${OBJECT_${SIMD}} PRIVATE
-      ENABLE_${SIMD}=1
-      DORENAME=1
-      ${COMMON_TARGET_DEFINITIONS}
-      )
-    add_dependencies(${OBJECT_${SIMD}} rename${SIMD}.h.generated)
+    add_library(${OBJECTsp_${SIMD}} OBJECT sleefsimdsp.c ${HEADER_${SIMD}})    
+    if (INDEX_ALIAS EQUAL -1)
```

Comment:
I think we can simplify this. We should always generate the alias header file, just use the `ENABLE_ALIAS` macro in the `mkalias` tool sources. If it set to one, generated the alias, otherwise leave the file empty. This would make possible to remove this extra logic that you have added here.

Reply:
I don't fully understand what you mean, but the alias is only generated if a dispatcher is not available. For example, Sleef_sind4_u10 calls a dispatcher, while Sleef_sind8_u10 is an alias of Sleef_sind8_u10avx512f.

Currently, ENABLE_ALIAS macro is used to indicate if the compiler supports weak alias.

Reply:
Apologies, I should have make it more clear. I think you can avoid adding such logic in the cmake configuration.

I don't think it is necessary to split the configuration for single precision and double precision. You have done this because you have decided to simplify the mkalias tool to accept only SP or only DP parameters, but his is not what we have done for other `mk*` tools. Tools like `mkrename` accept at once all the SP and DP parameters, it is then the `ENABLE_DP` and `ENABLE_SP` macros in the code of the library that decide what to load.